src/core/connectivity.jl

```diff
@@ -213,20 +220,84 @@ struct LatticeConnectivity <: AbstractConnectivity
 
         @assert +(qubits_per_row...) == qubit_count "Failed to build lattice"
 
-        new(qubits_per_row, (nrows, ncols), excluded_positions)
+        sorted_connections =
+            get_sorted_excluded_connections_for_lattice(nrows, ncols, excluded_connections)
+
+        new(qubits_per_row, (nrows, ncols), excluded_positions, sorted_connections)
     end
 
     function LatticeConnectivity(
         qubits_per_row::Vector{Int},
         excluded_positions = Vector{Int}(),
+        excluded_connections = Vector{Tuple{Int,Int}}(),
     )
         nrows = maximum(qubits_per_row)
-        ncols = ceil(+(qubits_per_row...) / nrows)
+        ncols = ceil(Int, +(qubits_per_row...) / nrows)
+
+        sorted_connections =
+            get_sorted_excluded_connections_for_lattice(nrows, ncols, excluded_connections)
 
-        new(qubits_per_row, (nrows, ncols), excluded_positions)
+        new(qubits_per_row, (nrows, ncols), excluded_positions, sorted_connections)
     end
 end
 
+function get_sorted_excluded_connections_for_lattice(
+    nrows::Int,
+    ncols::Int,
+    excluded_connections::Vector{Tuple{Int,Int}},
+)::Vector{Tuple{Int,Int}}
+
+    num_connections = length(excluded_connections)
+    sorted_connections = Vector{Tuple{Int,Int}}(undef, num_connections)
+    for (i_connection, connection) in enumerate(excluded_connections)
+        if connection[1] == connection[2]
+            throw(AssertionError("connection $connection must connect to different qubits"))
+        end
+
+        if connection[1] > connection[2]
+            sorted_connection = (connection[2], connection[1])
+        else
+            sorted_connection = connection
+        end
+
+        if sorted_connection[1] < 1
+            throw(
+                AssertionError(
+                    "connection $connection must have qubits with indices greater than 0",
+                ),
+            )
+        end
+
+        num_qubits = nrows * ncols
+        if sorted_connection[2] > num_qubits
+            throw(
+                AssertionError(
+                    "connection $connection must have qubits with indices less than" *
+                    " $(num_qubits + 1)",
+                ),
+            )
+        end
+
+        first_qubit_row_index = ((sorted_connection[1] - 1) ÷ ncols) + 1
+        near_index = sorted_connection[1] + ncols
+        coupler_exists =
+            sorted_connection[2] == near_index ||
+            (isodd(first_qubit_row_index) && sorted_connection[2] == near_index + 1) ||
```

Comment:
this logic only works for some LatticeDimensions and not others. Snowflurry.AnyonYamaskaConnectivity for instance does not follow this logic. Let's discuss how to address this offline

Reply:
changing the layout generation (https://github.com/SnowflurrySDK/Snowflurry.jl/pull/422) solves this problem

Reply:
The logic now works with the new layout.